core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -543,21 +543,22 @@ private[spark] class TaskSetManager(
         case TaskLocality.NO_PREF => pendingTasksWithNoPrefs.nonEmpty
         case TaskLocality.RACK_LOCAL => moreTasksToRunIn(pendingTasksForRack)
       }
+      val previousLocalityIndex = currentLocalityIndex
       if (!moreTasks) {
         // This is a performance optimization: if there are no more tasks that can
         // be scheduled at a particular locality level, there is no point in waiting
         // for the locality wait timeout (SPARK-4939).
         lastLaunchTime = curTime
-        logDebug(s"No tasks for locality level ${myLocalityLevels(currentLocalityIndex)}, " +
-          s"so moving to locality level ${myLocalityLevels(currentLocalityIndex + 1)}")
         currentLocalityIndex += 1
+        logDebug(s"No tasks for locality level ${myLocalityLevels(previousLocalityIndex)}, " +
+          s"so moving to locality level ${myLocalityLevels(currentLocalityIndex)}")
       } else if (curTime - lastLaunchTime >= localityWaits(currentLocalityIndex)) {
         // Jump to the next locality level, and reset lastLaunchTime so that the next locality
         // wait timer doesn't immediately expire
         lastLaunchTime += localityWaits(currentLocalityIndex)
         currentLocalityIndex += 1
         logDebug(s"Moving to ${myLocalityLevels(currentLocalityIndex)} after waiting for " +
-          s"${localityWaits(currentLocalityIndex)}ms")
+          s"${localityWaits(previousLocalityIndex)}ms")
```

Comment:
Ah OK, not quite what I meant. But, how about a slightly different idea. To match how the debug statement above works, leave it as-is, but then make this one:

```
logDebug(s"Moving to ${myLocalityLevels(currentLocalityIndex)} after waiting for " +		          
  s"${localityWaits(currentLocalityIndex+1)}ms")
currentLocalityIndex += 1
```